File: cloud/internal/handler/baseline.go
package handler

import (
	"net/http"

	"github.com/go-chi/chi/v5"
	"github.com/google/uuid"
	"github.com/kx0101/replayer-cloud/internal/middleware"
)

func (h *Handler) SetBaseline(w http.ResponseWriter, r *http.Request) {
	userID := middleware.GetUserID(r.Context())

	idStr := chi.URLParam(r, "id")
	id, err := uuid.Parse(idStr)
	if err != nil {
		respondError(w, http.StatusBadRequest, "invalid run id")
		return
	}

	if err := h.store.SetBaselineForUser(r.Context(), userID, id); err != nil {
		if err.Error() == "run not found" {
			respondError(w, http.StatusNotFound, "run not found")
			return
		}
		respondError(w, http.StatusInternalServerError, "failed to set baseline")
		return
	}

	respondJSON(w, http.StatusOK, map[string]string{"status": "ok"})
}

File: crates/replayer-cloud/src/handler/baseline.rs
use axum::{extract::State, http::StatusCode, response::Response, Extension};
use uuid::Uuid;

use crate::middleware::AuthUserId;

use super::response::{respond_error, respond_json};
use super::AppState;

pub async fn set_baseline(
    State(state): State<AppState>,
    Extension(AuthUserId(user_id)): Extension<AuthUserId>,
    axum::extract::Path(id): axum::extract::Path<Uuid>,
) -> Response {
    if let Err(e) = state.store.set_baseline_for_user(user_id, id).await {
        let msg = e.to_string();
        if msg.contains("run not found") {
            return respond_error(StatusCode::NOT_FOUND, "run not found");
        }
        tracing::error!("error setting baseline: {}", e);
        return respond_error(StatusCode::INTERNAL_SERVER_ERROR, "failed to set baseline");
    }

    respond_json(StatusCode::OK, &serde_json::json!({"status": "ok"}))
}
